WordPress/Classes/ViewRelated/Media/MediaPicker/Helpers/MediaPickerMenuController.swift

```diff
@@ -0,0 +1,62 @@
+import Photos
+import PhotosUI
+
+final class MediaPickerMenuController: NSObject {
+    var onSelection: ((MediaPickerSelection) -> Void)?
+
+    fileprivate func didSelect(_ items: [MediaPickerItem], source: String) {
+        let selection = MediaPickerSelection(items: items, source: source)
+        DispatchQueue.main.async {
+            self.onSelection?(selection)
+        }
```

Comment:
Why dispatch async here? For chaining dismising presenting view controllers, maybe?

Reply:
If that's the cause, maybe we can use `dismiss(animated:completion:)` in the call sites, which would be syntactically nicer than dispatch async?

Reply:
I was debating this, but the problem with `dismiss(animated:completion:)` is that it you perform `onSelection` in the `completion` callback, it feels too slow. `DispatchQueue.main.async` is probably redundant here, but I added it just in case.